src/libraries/System.Text.Json/tests/Serialization/ExtensionDataTests.cs

```diff
@@ -1066,5 +1066,53 @@ public override void Write(Utf8JsonWriter writer, JsonElement value, JsonSeriali
                 writer.WriteEndObject();
             }
         }
+
+        [Theory]
+        [InlineData(typeof(HasExtensionData<Dictionary<string, JsonElement>>))]
+        [InlineData(typeof(HasExtensionData<SortedDictionary<string, JsonElement>>))]
+        [InlineData(typeof(HasExtensionData<IDictionary<string, JsonElement>>))]
+        [InlineData(typeof(HasExtensionData<IReadOnlyDictionary<string, JsonElement>>))]
+        [InlineData(typeof(HasExtensionData<ImmutableDictionary<string, JsonElement>>))]
+        [InlineData(typeof(HasExtensionData<ImmutableSortedDictionary<string, JsonElement>>))]
+        [InlineData(typeof(HasExtensionData<IImmutableDictionary<string, JsonElement>>))]
+        [InlineData(typeof(HasExtensionData<ConcurrentDictionary<string, JsonElement>>))]
+        public static void TypeValidityAsExtensionDataWithJsonElementValue(Type type)
+        {
+            const string json = "{\"foo\": 123}";
+
+            object obj = JsonSerializer.Deserialize(json, type);
+            IDictionary<string, JsonElement> extData = (IDictionary<string, JsonElement>)type.GetProperty("ExtensionData").GetValue(obj)!;
+            Assert.True(extData.ContainsKey("foo"));
+
+            int value = extData["foo"].GetInt32();
+            Assert.Equal(123, value);
+        }
+
+        [Theory]
+        [InlineData(typeof(HasExtensionData<Dictionary<string, object>>))]
+        [InlineData(typeof(HasExtensionData<SortedDictionary<string, object>>))]
+        [InlineData(typeof(HasExtensionData<IDictionary<string, object>>))]
+        [InlineData(typeof(HasExtensionData<IReadOnlyDictionary<string, object>>))]
+        [InlineData(typeof(HasExtensionData<ImmutableDictionary<string, object>>))]
+        [InlineData(typeof(HasExtensionData<ImmutableSortedDictionary<string, object>>))]
+        [InlineData(typeof(HasExtensionData<IImmutableDictionary<string, object>>))]
+        [InlineData(typeof(HasExtensionData<ConcurrentDictionary<string, object>>))]
+        public static void TypeValidityAsExtensionDataWithObjectValue(Type type)
```

Comment:
The test for converters are handled separately from these additional tests such as the test `CustomObjectConverterInExtensionProperty`. We could add more tests for the specific types here but that wouldn't be testing any new codepaths since there's no specific logic for converters changed here so these verify we didn't break anything existing